src/com/google/ProductionInstrumentationReporter.java

```diff
@@ -0,0 +1,176 @@
+          + "this reporter.")
+  private String finalResultOutput = "";
+
+
+  public static void main(String[] args) {
+    try {
+      new ProductionInstrumentationReporter().doMain(args);
+    } catch (IOException e) {
+      System.err.println(e.getMessage());
+      System.err.println(e.getStackTrace());
+    }
+  }
+
+  /**
+   * This function reads a file at the given filePath and converts the contents into a string.
+   */
+  public static String readFile(String filePath) throws IOException {
+    return CharStreams.toString(new FileReader(filePath));
+  }
+
+  /**
+   * Reads all files found in folder and converts the contents of each file to a Map<String,
+   * ProfilingData> data structure where it is a mapping of the unique param value to the encoded
+   * values. The folder contains all the reports sent by the instrumented production code.
+   */
+  private List<Map<String, ProfilingData>> getAllExecutionResults(File folder)
+      throws IOException {
+    List<Map<String, ProfilingData>> result = new ArrayList<>();
+
+    for (final File fileEntry : folder.listFiles()) {
+      String executionResult = readFile(fileEntry.getAbsolutePath());
+      Type type = new TypeToken<Map<String, ProfilingData>>() {
+      }.getType();
+      Map<String, ProfilingData> executedInstrumentationData = new Gson()
+          .fromJson(executionResult, type);
+
+      result.add(executedInstrumentationData);
+
+    }
+
+    return result;
+  }
+
+  /**
+   * Creates a file with the given fileName (including extension) with the contents of the file
+   * being provided by fileContents.
+   */
+  private void createFile(String fileName, String fileContents) throws IOException {
+
+    File fold = new File(fileName);
+    fold.delete();
+    File myObj = new File(fileName);
+    myObj.createNewFile();
+
+    FileWriter myWriter = new FileWriter(fileName);
+    myWriter.write(fileContents);
+    myWriter.close();
+  }
+
+  public void doMain(String[] args) throws IOException {
+
+    parseCmdLineArguments(args);
+
+    InstrumentationMapping instrumentationMapping = InstrumentationMapping
+        .parse(instrumentationMappingLocation);
+
+    File folder = new File(instrumentationReportsDirectory);
+
+    List<Map<String, ProfilingData>> listOfExecutionResults = getAllExecutionResults(folder);
+
+    ProfilingReport profilingReport = createProfilingReport(
+        instrumentationMapping, listOfExecutionResults);
+
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    createFile(finalResultOutput, gson.toJson(profilingReport));
+
+  }
+
+  private void parseCmdLineArguments(String[] args) {
+    CmdLineParser parser = new CmdLineParser(this);
+    parser.setUsageWidth(80);
+    try {
+      parser.parseArgument(args);
+    } catch (CmdLineException e) {
+      System.err.println(e.getMessage());
+      return;
+    }
+
+    if (Strings.isNullOrEmpty(instrumentationMappingLocation)) {
```

Comment:
Assuming the Closure Compiler version of args4j is high enough, would marking these options "required" replace these nullOrEmpty checks?

https://args4j.kohsuke.org/apidocs/org/kohsuke/args4j/Option.html#required()

Reply:
Removed nullOrEmpty checks with required. I am using the same version of args4j as closure so should be fine.